app/src/main/java/com/example/footballmanager/screens/result_screens/success_components/MatchScoreTab.kt

```diff
@@ -0,0 +1,324 @@
+import androidx.compose.foundation.layout.requiredSize
+import androidx.compose.foundation.layout.requiredWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.rememberAsyncImagePainter
+import com.example.footballmanager.R
+import com.example.footballmanager.network.FixtureDataWrapper
+import com.example.footballmanager.screens.view_models.HomeViewModel
+import java.time.LocalDate
+import kotlin.math.log
+
+
+@Composable
+fun MatchScoreTab(
+    fixture: FixtureDataWrapper,
+    modifier: Modifier = Modifier,
+    homeViewModel: HomeViewModel = viewModel()
+) {
+    var lastLoadedAmount by remember {
+        mutableIntStateOf(10)
+    }
+    val lazyListState = rememberLazyListState()
+    val currentState = remember { derivedStateOf { lazyListState.firstVisibleItemIndex } }
+    val shouldLoadMore = remember { derivedStateOf { currentState.value + 10 >= lastLoadedAmount } }
+
+    LaunchedEffect(key1 = shouldLoadMore.value) {
+        lastLoadedAmount += 10
+    }
+
+    Column(
+        verticalArrangement = Arrangement.spacedBy(
+            dimensionResource(id = R.dimen.column_spacing),
+            Alignment.Top
+        ),
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = dimensionResource(id = R.dimen.column_padding))
+    ) {
+        val defaultValue = stringResource(id = R.string.default_value)
+        LazyColumn(
+            state = lazyListState,
+            modifier = Modifier
+                .fillMaxWidth()
+        ) {
+            var previousLeagueName = defaultValue
+            for (i in 0..lastLoadedAmount) {
+                val item = fixture.responseBody[i]
+                val leagueName = item.league?.name ?: defaultValue
```

Comment:
You could use `with` here. Read about [scope functions](https://kotlinlang.org/docs/scope-functions.html)

Reply:
You can use `with(item)` so that you don't have to refer to the item each time